src/components/Group/Group.js

```diff
@@ -5,85 +5,96 @@ import Task from '../Task/Task';
+import { fetchDeleteTask, fetchUpdateGroup, fetchUpdateTask } from '../../redux/slices/groups';
 
 const ListGroup = ({
-  group, 
-  groups, 
-  setGroups, 
+  group,
+  setStatistics,
   removeGroup, 
 }) => {
+    const groupId = group._id;
+    const dispatch = useDispatch();
     const [edit, setEdit] = useState(null);
     const [value, setValue] = useState('');
     const [modalTaskVisible, setModalTaskVisible] = useState(false);
-    const checkCompleted = () => {
+    const checkCompleted = (group) => {
         let completed = 0;
         let notCompleted = 0;
         group.tasks.map((i) => i.status ? ++notCompleted : ++completed);
-    }
-    
-    const addTask = (newTask) => {
-        group.tasks.push(newTask);
-        setGroups([...groups]);
-        setModalTaskVisible(false);
+        const updatedGroup = {
+            ...group,
+            completed,
+            notCompleted,
+          };
+        dispatch(fetchUpdateGroup({updatedGroup, groupId}));
+        setStatistics(updatedGroup);
     }
     
     const removeTask = (task) => {
-        let i = group.tasks.indexOf(task);
-        if(i >= 0) {
-            group.tasks.splice(i,1);
-        }
-        setGroups([...groups]);
+        const taskId = task._id
+        dispatch(fetchDeleteTask({groupId, taskId}));
     }
     
     const statusTask = (task) => {
-        group.tasks.filter( i => {
-            if (i.id === task.id) i.status = !i.status;
-            return i;
-          }
-        )
-        setGroups([...groups]);
+        const updatedTask = {
+            ...task,
+            status: !task.status
+          };
+        const taskId = task._id
+        dispatch(fetchUpdateTask({updatedTask, groupId, taskId}));
     }
 
     const editTask = (task) => {
-        setEdit(task.id);
+        setEdit(task._id);
         setValue(task.title);
     }
     
     const saveTask = (task) => {
-        group.tasks.filter( i => {
-            if (i.id === task.id) i.title = value;
-            setEdit(null);
-            setValue('');
-            return i;
-        }
-        )
-        setGroups([...groups]);
+        const updatedTask = {
+            ...task,
+            title: value
+          };
+        const taskId = task._id
+        setEdit(null);
+        setValue('');
+        dispatch(fetchUpdateTask({updatedTask, groupId, taskId}));
+    }
+
+    const priorityTask = (task) => {
+        const updatedTask = {
+            ...task,
+            priority: !task.priority
+          };
+        const taskId = task._id
+        dispatch(fetchUpdateTask({updatedTask, groupId, taskId}));
     }
     return (
         <div>
+            <div>
+                <Link className={styles.link} to='/statistics/'>
+                    <MyButton onClick={() => checkCompleted(group)} >Статистика групи</MyButton>
+                </Link>
+            </div>
             <h1 className={styles.groupHeader}>{group.title}</h1>
```

Comment:
After destructuring:
```suggestion
            <h1 className={styles.groupHeader}>{groupTitle}</h1>
```